frontend-wasm/src/code_translator/mod.rs

```diff
@@ -0,0 +1,1035 @@
+            let block = {
+                let i = state.control_stack.len() - 1 - (depth as usize);
+                let frame = &mut state.control_stack[i];
+                frame.set_branched_to_exit();
+                frame.br_destination()
+            };
+            data.push((depth, block));
+        }
+        let def_block = {
+            let i = state.control_stack.len() - 1 - (default as usize);
+            let frame = &mut state.control_stack[i];
+            frame.set_branched_to_exit();
+            frame.br_destination()
+        };
+        builder.ins().switch(val, data, def_block, span);
+    } else {
+        // Here we have jump arguments, but Midens's switch op doesn't support them
+        // We then proceed to split the edges going out of the br_table
+        let return_count = jump_args_count;
+        let mut dest_block_sequence = vec![];
+        let mut dest_block_map = HashMap::new();
+        for depth in targets.targets() {
+            let depth = depth?;
+            let branch_block = match dest_block_map.entry(depth as usize) {
+                hash_map::Entry::Occupied(entry) => *entry.get(),
+                hash_map::Entry::Vacant(entry) => {
+                    let block = builder.create_block();
+                    dest_block_sequence.push((depth as usize, block));
+                    *entry.insert(block)
+                }
+            };
+            data.push((depth, branch_block));
+        }
+        let default_branch_block = match dest_block_map.entry(default as usize) {
+            hash_map::Entry::Occupied(entry) => *entry.get(),
+            hash_map::Entry::Vacant(entry) => {
+                let block = builder.create_block();
+                dest_block_sequence.push((default as usize, block));
+                *entry.insert(block)
+            }
+        };
+        builder.ins().switch(val, data, default_branch_block, span);
+        for (depth, dest_block) in dest_block_sequence {
+            builder.switch_to_block(dest_block);
+            builder.seal_block(dest_block);
+            let real_dest_block = {
+                let i = state.control_stack.len() - 1 - depth;
+                let frame = &mut state.control_stack[i];
+                frame.set_branched_to_exit();
+                frame.br_destination()
+            };
+            let destination_args = state.peekn_mut(return_count);
+            builder.ins().br(real_dest_block, destination_args, span);
+        }
+        state.popn(return_count);
+    }
+    state.reachable = false;
+    Ok(())
+}
+
+/// Return the total Miden VM memory size (2^32 addresses * word (4 felts) bytes) in 64KB pages
+const fn mem_total_pages() -> i32 {
+    let bytes_fit_in_felt = 4; // although more than 32 bits can fit into a felt, use 32 bits to be safe
+    let felts_in_word = 4;
+    let total_addresses = u32::MAX as i64;
+    (total_addresses * (felts_in_word * bytes_fit_in_felt) / (64 * 1024)) as i32
```

Comment:
*Suggestion*: This whole thing could be a constant, rather than a const function, but it probably will go away as soon as the `memory_size` instruction is added anyway, so your call

```suggestion
    const FELT_BYTES: u64 = 4; // felts are effectively 32 bits
    const WORD_BYTES: u64 = 4 * FELT_BYTES; // 4 felts per word
    const PAGE_SIZE: u64 = 64 * 1024;
    const MEMORY_SIZE: u64 = (u32::MAX as u64 * WORD_BYTES);
    MEMORY_SIZE / PAGE_SIZE
```

Reply:
Thanks! I like yours more. It's more readable. I took it but left it inside the const function to keep it contained. It's a temporary solution anyway.